plugins/source/aws/scripts/policy_docs/main.go

```diff
@@ -61,28 +67,58 @@ func newPolicyInfo(name string, queries []Query, allTables []Table) *PolicyInfo
 
 // Tables returns the unique set of tables needed to run all queries
 func (pi *PolicyInfo) setTables(allTables Tables) {
-	t := map[string]struct{}{}
+	t, v := map[string]bool{}, map[string]bool{}
+	pi.CreatedViews = map[string]bool{}
+
+	// Add all views detected from CREATE VIEW statements
 	for _, q := range pi.Queries {
+		if q.View {
+			pi.CreatedViews[q.ViewName] = true
+		}
+	}
+
+	// Accessed views from queries, should be a subset of the above
+	for _, q := range pi.Queries {
+		for _, vi := range q.Views {
+			v[vi] = true
+		}
+	}
+
+	for _, q := range pi.Queries {
+		// Add all tables from queries *and* views
 		for _, table := range q.Tables {
-			t[table] = struct{}{}
+			if pi.CreatedViews[table] {
+				// This is a view, not a table
+				v[table] = true
+			}
+
+			t[table] = true
 			ancestors := allTables.FindAncestors(table)
 			for _, a := range ancestors {
-				t[a.Name] = struct{}{}
+				t[a.Name] = true
```

Comment:
better leave as it was (bool uses mem, empty struct doesn't)

Reply:
Doesn't matter and this makes the template simpler (or, usable)